diff --git a/node_modules/cddl/build/parser.js b/node_modules/cddl/build/parser.js
index f293154..9070e1c 100644
--- a/node_modules/cddl/build/parser.js
+++ b/node_modules/cddl/build/parser.js
@@ -26,15 +26,18 @@ export default class Parser {
     l;
     curToken = NIL_TOKEN;
     peekToken = NIL_TOKEN;
+    peekBelowToken = NIL_TOKEN;
     constructor(filePath) {
         this.#filePath = filePath;
         this.l = new Lexer(fs.readFileSync(filePath, 'utf-8'));
         this.nextToken();
         this.nextToken();
+        this.nextToken();
     }
     nextToken() {
         this.curToken = this.peekToken;
-        this.peekToken = this.l.nextToken();
+        this.peekToken = this.peekBelowToken;
+        this.peekBelowToken = this.l.nextToken();
         return true;
     }
     parseAssignments() {
@@ -110,7 +113,10 @@ export default class Parser {
          *       attireBlock
          *   )
          */
-        if (closingTokens.length && this.peekToken.Type === Tokens.SLASH) {
+        if (closingTokens.includes(Tokens.RPAREN) &&
+            this.peekToken.Type === Tokens.SLASH &&
+            this.peekBelowToken.Type !== Tokens.SLASH &&
+            !(this.curToken.Type === Tokens.SLASH && this.peekToken.Type === Tokens.SLASH)) {
             const propertyType = [];
             while (!closingTokens.includes(this.curToken.Type)) {
                 propertyType.push(...this.parsePropertyTypes());
@@ -123,6 +129,9 @@ export default class Parser {
                     this.nextToken();
                 }
             }
+            if (this.curToken.Type === Tokens.RPAREN) {
+                this.nextToken();
+            }
             if (groupName) {
                 const variable = {
                     Type: 'variable',
@@ -163,6 +172,23 @@ export default class Parser {
             return [prop];
         }
         while (!closingTokens.includes(this.curToken.Type)) {
+            /**
+             * check if we have a group choice instead of an assignment
+             */
+            if (this.curToken.Type === Tokens.SLASH && this.peekToken.Type === Tokens.SLASH) {
+                if (valuesOrProperties.length === 0) {
+                    throw this.parserError('Unexpected group choice operator "//" at start of group');
+                }
+
+                if (!isChoice) {
+                    const last = valuesOrProperties.pop();
+                    valuesOrProperties.push([last]);
+                    isChoice = true;
+                }
+                this.nextToken();
+                this.nextToken();
+                continue;
+            }
             const propertyType = [];
             const comments = [];
             let isUnwrapped = false;
@@ -197,13 +223,24 @@ export default class Parser {
                 this.curToken.Literal === Tokens.LBRACK ||
                 this.curToken.Literal === Tokens.LPAREN) {
                 const innerGroup = this.parseAssignmentValue();
-                valuesOrProperties.push({
+                const prop = {
                     HasCut: false,
                     Occurrence: occurrence,
                     Name: '',
                     Type: innerGroup,
                     Comments: []
-                });
+                };
+
+                if (isChoice) {
+                    valuesOrProperties[valuesOrProperties.length - 1].push(prop);
+                } else {
+                    valuesOrProperties.push(prop);
+                }
+
+                if (this.curToken.Type === Tokens.COMMA) {
+                    this.nextToken();
+                    isChoice = false;
+                }
                 continue;
             }
             /**
@@ -317,7 +354,7 @@ export default class Parser {
              */
             const props = this.parseAssignmentValue();
             let operator = this.isOperator() ? this.parseOperator() : undefined;
-            if (!isChoice && this.curToken.Type === Tokens.SLASH) {
+            if (!isChoice && this.curToken.Type === Tokens.SLASH && this.peekToken.Type !== Tokens.SLASH) {
                 this.nextToken();
                 const nextType = this.parsePropertyType();
                 if (Array.isArray(props)) {
@@ -485,10 +522,6 @@ export default class Parser {
     openSegment() {
         if (this.curToken.Type === Tokens.LBRACE) {
             this.nextToken();
-            if (this.peekToken.Type === Tokens.LPAREN) {
-                this.nextToken();
-                return [Tokens.RPAREN, Tokens.RBRACE];
-            }
             return [Tokens.RBRACE];
         }
         else if (this.curToken.Type === Tokens.LPAREN) {
@@ -549,6 +582,9 @@ export default class Parser {
                         Unwrapped: isUnwrapped
                     };
                 }
+                else if (this.curToken.Literal === Tokens.LBRACE) {
+                    type = this.parseAssignmentValue();
+                }
                 else if (this.curToken.Type === Tokens.IDENT) {
                     type = {
                         Type: 'group',
